test/e2e/syncer/syncer_test.go

```diff
@@ -251,6 +255,76 @@ func TestSyncerLifecycle(t *testing.T) {
+		podsAllRole := &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{Name: "pods-all"},
+			Rules: []rbacv1.PolicyRule{
+				rbachelper.NewRule("*").Groups("").Resources("pods").RuleOrDie(),
+				rbachelper.NewRule("*").Groups("").Resources("pods/log").RuleOrDie(),
+				rbachelper.NewRule("*").Groups("").Resources("pods/exec").RuleOrDie(),
+			},
+		}
+
+		_, err = upstreamKubeClusterClient.RbacV1().Roles(upstreamNamespace.Name).Create(logicalcluster.WithCluster(ctx, wsClusterName), podsAllRole, metav1.CreateOptions{})
+		require.NoError(t, err, "failed to create upstream role")
+		podsAllRoleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{Name: "pods-all"},
+			Subjects: []rbacv1.Subject{
+				{Kind: "ServiceAccount", Name: syncerFixture.SyncerID},
+			},
+			RoleRef: rbacv1.RoleRef{Kind: "ClusterRole", Name: "pods-all"},
+		}
+
+		_, err = upstreamKubeClusterClient.RbacV1().RoleBindings(upstreamNamespace.Name).Create(logicalcluster.WithCluster(ctx, wsClusterName), podsAllRoleBinding, metav1.CreateOptions{})
+		require.NoError(t, err, "failed to create upstream rolebinding")
+
+		// KCP will forwards all requests to the downstream cluster
+		// kubectl --server=https://{host}/clusters/{cluster}/services/tunnels/syncer-proxy/{syncer-name} get pods -A
+		t.Logf("Get logs through KCP from downstream deployment")
+		u, err := url.Parse(syncerFixture.SyncerConfig.UpstreamConfig.Host)
+		require.NoError(t, err, "failed to parse upstream Host for syncer")
+		proxiedConfig := rest.CopyConfig(syncerFixture.SyncerConfig.UpstreamConfig)
+		proxiedConfig.Host = fmt.Sprintf("%s://%s/clusters/%s/services/tunnels/syncer-proxy/%s", u.Scheme, u.Host, wsClusterName, syncerFixture.SyncerConfig.SyncTargetName)
+
+		proxiedKcpClient, err := kubernetesclientset.NewForConfig(proxiedConfig)
+		require.NoError(t, err)
+
+		framework.Eventually(t, func() (bool, string) {
+			deployment, err = proxiedKcpClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, iccUpstreamDeployment.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				return false, ""
+			}
+			require.NoError(t, err)
+			pods, err := proxiedKcpClient.CoreV1().Pods(downstreamNamespaceName).List(ctx, metav1.ListOptions{})
+			if apierrors.IsUnauthorized(err) {
+				return false, ""
+			}
+			require.NoError(t, err)
+
+			var podLogs bytes.Buffer
+			for _, pod := range pods.Items {
+				for _, c := range pod.Spec.Containers {
+					request := proxiedKcpClient.CoreV1().RESTClient().Get().
+						Resource("pods").
+						Namespace(pod.Namespace).
+						Name(pod.Name).SubResource("log").
+						Param("container", c.Name)
+					logs, err := request.Do(context.TODO()).Raw()
+					if err != nil {
+						klog.Errorf("Failed to get logs for pod %s/%s container %s: %v", pod.Namespace, pod.Name, c.Name, err)
+						continue
+					}
+					for _, line := range strings.Split(string(logs), "\n") {
+						t.Logf("Pod %s/%s container %s: %s", pod.Namespace, pod.Name, c.Name, line)
```

Comment:
will this spam the test logs? How many lines will we see? How many containers?

Reply:
I've copied/pasted from the existing `podLogs` function, there is only one pod with no many logs

Reply:
let me change it